lib/internal/streams/legacy.js

```diff
@@ -111,4 +112,24 @@ function prependListener(emitter, event, fn) {
     emitter._events[event] = [fn, emitter._events[event]];
 }
 
+let defaultHighwaterMark = null;
+ObjectDefineProperty(Stream, 'defaultHighwaterMark', {
+  get () {
+    return defaultHighwaterMark || Buffer.poolSize || 16 * 1024;
```

Comment:
1. Setting it to `0` sets it to `Buffer.poolSize` instead. That is undocumented.
2. `16 * 1024` fallback is undocumented.
3. This __decreases__ the default hwm value from `16 * 1024` to `8 * 1024`. How will this affect performance?

Perhaps, it would make sense to just remove the relationship to `Buffer.poolSize` and set the default to `16 * 1024`?
Also, not here, but on the `let defaultHighwaterMark` line.